docs/operations.md

```diff
@@ -28,6 +28,138 @@ Clients may inspect this specification at run-time so to obey the various limits
 
+### Job Events
+
+Job events provide a notification mechanism that informs clients about certain operations happening on jobs.
+This approach eliminates the need for clients to continuously poll wfx, thus optimizing network usage and client resources.
+Job events can be useful for user interfaces (UIs) and other applications that demand near-instantaneous updates.
+
+#### Architecture
+
+The job events notification system leverages [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events) (SSE) for its implementation.
+Below is a high-level overview of how the communication flow operates:
+
+```txt
+ ┌────────┐                            ┌─────┐
+ │ Client │                            │ wfx │
+ └────────┘                            └─────┘
+     |                                    |
+     |        HTTP GET /jobs/events       |
+     |-----------------------------------►|
+     |                                    |
+     |                                    |
+     |           Event Loop               |
+ ┌───|────────────────────────────────────|───┐
+ │   |  [Content-Type: text/event-stream] |   │
+ │   |                                    |   │
+ │   |                                    |   │
+ │   |            Push Event              |   │
+ │   |◄-----------------------------------|   │
+ │   |                                    |   │
+ └───|────────────────────────────────────|───┘
+     |                                    |
+     ▼                                    ▼
+ ┌────────┐                            ┌─────┐
+ │ Client │                            │ wfx │
+ └────────┘                            └─────┘
+```
+
+1. The client initiates communication by sending an HTTP `GET` request to the `/jobs/events` endpoint. Clients may also
+   include optional [filter parameters](#filter-parameters) within the request.
+2. Upon receipt of the request, `wfx` sets the `Content-Type` header to `text/event-stream`.
+3. The server then initiates a stream of job events in the response body, allowing clients to receive instant updates.
+
+#### Event Format Specification
+
+The event stream looks as follows:
+
+```
+data: [...]
+id: 1
+
+data: [...]
+id: 2
+
+[...]
+```
+
+with a single event being structured as follows:
+
+```
+data: { "action": "<ACTION>", "job": <JOB> }
+id: <EVENT_ID>
+
+```
+
+**Note**: Each event is terminated by a pair of newline characters `\n\n` (as required by the SSE spec).
+
+The semantics of the individual fields is:
+
+- `<ACTION>` specifies the type of event that occurred. The valid actions are:
+  - `CREATE`: a new job has been created
+  - `DELETE`: an existing job has been deleted
+  - `ADD_TAGS`: tags were added to a job
+  - `DELETE_TAGS`: tags were removed from a job
+  - `UPDATE_STATUS`: job status has been updated
+  - `UPDATE_DEFINITION`: job definition has been updated
+- `<JOB>` is the JSON representation of the job associated with this event
```

Comment:
This shouldn't be the whole job JSON but instead sub-trees of it containing the relevant data related to `ACTION`, e.g., I'm only interested in `status:`  when I receive an `UPDATE_STATUS` event.

Reply:
And the client id, since we also allow filtering by client id. I have changed the code and updated the documentation.